src/modules/vendor/components/programs/ProgramCreationModal.vue

```diff
@@ -58,6 +58,9 @@ export default {
     submit () {
       this.$emit('submit');
     },
+    update (event, prop) {
```

Comment:
je me dit qu'on pourrait essayer de généraliser cette méthode et la mettre dans utils par exemple.. ca permettrait de moins nous fatiguer lors de la refacto et de ne pas recrire 100x la même logique..
le problème c'est que ici on écrit en dur `newProgram` donc il faudrait soit passer le nom de l'objet en 3e paramètre ou alors garder 2 paramètres avec quelque chose comme ça:
```
update (event, prop) {
      this.$emit(`update:${prop.split('.')[0]}`, { ...this.newProgram, [prop.split('.')[1]]: event });
},
```
enfaite ici le paramètre prop attend une string de type "nomObjet.nomPropriété" par exemple`@input="update($event, 'newProgram.category')"`
je ne sais pas si je suis très clair..

Reply:
si si c'est clair :)

en voulant faire ca, tu evite de reecrire la meme methode dans plusieurs div mais en parallele tu alourdis le html et notamment tous les `@input` ou tu devras ajouter le nom de la props en plus ce qui le rendra moins lisible

Reply:
d'accord, oui tu as raison!
(par curiosité, par une mixin ça allègerait un peu plus? )

une version sans le `this.newProgram` laissé :)
```
update (event, prop) {
this.$emit(`update:${prop.split('.')[0]}`, { ...this[prop.split('.')[0]], [prop.split('.')[1]]: event });
},
```

Reply:
ca alegerait sans doute un peu oui :)
ca fonctionne le `this[object]` ?

Reply:
yes :)